temporalio/worker/_workflow_instance.py

```diff
@@ -1714,12 +1731,11 @@ def _register_task(
         name: Optional[str],
     ) -> None:
         self._assert_not_read_only("create task")
-        # Name not supported on older Python versions
-        if sys.version_info >= (3, 8):
-            # Put the workflow info at the end of the task name
-            name = name or task.get_name()
-            name += f" (workflow: {self._info.workflow_type}, id: {self._info.workflow_id}, run: {self._info.run_id})"
-            task.set_name(name)
+
+        # Put the workflow info at the end of the task name
+        name = name or task.get_name()
+        name += f" (workflow: {self._info.workflow_type}, id: {self._info.workflow_id}, run: {self._info.run_id})"
+        task.set_name(name)
```

Comment:
I know in theory we only support >=3.8 but just to confirm -- we are breaking any <3.8 users here?

Reply:
I believe so.
Though, even 3.8 is reaching EOL in a few months (2024-10) so I think that it's safe to do at this point.

Reply:
I think the question is why change this code at all? This seems completely unrelated to the current PR. If we want a task to come back and remove our pre-3.8 code we can create it.

Reply:
Still not sure this code should have changed

Reply:
I tried reverting it to how it read before, let me know if I did it wrong :(